indexer/scripts/elastic-conf.py

```diff
@@ -96,6 +116,55 @@ def main_loop(self) -> None:
             logger.error("One or more configurations failed. Check logs for details.")
             return
 
+    def repository_exists(self, es: Elasticsearch) -> bool:
+        repo_name = self.es_snapshot_repo
+        try:
+            response = es.snapshot.get_repository(name=repo_name)
+            return repo_name in response
+        except Exception:
+            logger.exception("Error validating repository: %s", repo_name)
+        return False
+
+    def register_repository(self, es: Elasticsearch) -> bool:
+        status = False
+        repo_type = self.es_snapshot_repo_type
+        if not repo_type:
+            logger.error("Repository type must be either 'fs' or 's3'")
+            return status
```

Comment:
```suggestion
        repo_type = self.es_snapshot_repo_type
        if repo_type not in ["fs", "s3"]:
            logger.error("Unsupported repository type: '%s'. Must be either 'fs' or 's3'", repo_type)
            return False
```

This should make sure _repo_name_ is set **and** it's one of the supported type. Don't think we need _status_.